app/src/test/java/org/fossasia/openevent/general/AppLinkUtilsTest.kt

```diff
@@ -0,0 +1,55 @@
+package org.fossasia.openevent.general
+
+import android.net.Uri
+import org.fossasia.openevent.general.utils.EVENT_IDENTIFIER
+import org.fossasia.openevent.general.utils.RESET_PASSWORD_TOKEN
+import org.fossasia.openevent.general.utils.AppLinkUtils
+import org.fossasia.openevent.general.utils.VERIFICATION_TOKEN
+import org.junit.Test
+import org.junit.Assert.assertEquals
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+private const val EVENT = "event"
+private const val RESET_PASSWORD = "resetPassword"
+private const val VERIFY_EMAIL = "verifyEmail"
+
+@RunWith(RobolectricTestRunner::class)
+class AppLinkUtilsTest {
+
+    private fun getAppLink(type: String): Uri {
+        return when (type) {
+            EVENT -> Uri.parse("https://eventyay.com/e/5f6d3feb")
+            RESET_PASSWORD -> Uri.parse("https://eventyay.com/reset-password?token=822980340478781748445098077144")
+            VERIFY_EMAIL -> Uri.parse("https://eventyay.com/verify?token=WyJsaXZlLmhhcnNoaXRAaG")
+            else -> Uri.parse("")
+        }
+    }
+
+    @Test
+    fun `should get event link`() {
+        val uri = getAppLink(EVENT)
+        assertEquals(R.id.eventDetailsFragment, AppLinkUtils.getDestinationId(uri))
+        assertEquals("""
+            5f6d3feb
+        """.trimIndent(), AppLinkUtils.getArguments(uri).getString(EVENT_IDENTIFIER))
+    }
+
+    @Test
+    fun `should get reset password link`() {
+        val uri = getAppLink(RESET_PASSWORD)
+        assertEquals(R.id.eventsFragment, AppLinkUtils.getDestinationId(uri))
+        assertEquals("""
+            822980340478781748445098077144
+        """.trimIndent(), AppLinkUtils.getArguments(uri).getString(RESET_PASSWORD_TOKEN))
+    }
+
+    @Test
+    fun `should get verify email link`() {
```

Comment:
I'm not sure if this is a good way to write a function name. Maybe ask the mentor about that.

Reply:
They have already written like that for test cases, these are under test implementation

Reply:
ok, it's new to me. Good to know about that